datasets/told-br/told-br.py

```diff
@@ -0,0 +1,178 @@
+
+# TODO: Add the licence for the dataset here if you can find it
+_LICENSE = "https://github.com/JAugusto97/ToLD-Br/blob/main/LICENSE_ToLD-Br.txt "
+
+# TODO: Add link to the official dataset URLs here
+# The HuggingFace Datasets library doesn't host the datasets but only points to the original files.
+# This can be an arbitrary nested dict/list of URLs (see below in `_split_generators` method)
+
+_URLS = {
+    "multilabel": "https://raw.githubusercontent.com/JAugusto97/ToLD-Br/main/ToLD-BR.csv",
+    "binary": "https://github.com/JAugusto97/ToLD-Br/raw/main/experiments/data/1annotator.zip",
+}
+
+
+class ToldBr(datasets.GeneratorBasedBuilder):
+    """Toxic/Abusive Tweets Classification Dataset for Brazilian Portuguese."""
+
+    VERSION = datasets.Version("1.0.0")
+
+    # This is an example of a dataset with multiple configurations.
+    # If you don't want/need to define several sub-sets in your dataset,
+    # just remove the BUILDER_CONFIG_CLASS and the BUILDER_CONFIGS attributes.
+
+    # If you need to make complex sub-parts in the datasets with configurable options
+    # You can create your own builder configuration class to store attribute, inheriting from datasets.BuilderConfig
+    # BUILDER_CONFIG_CLASS = MyBuilderConfig
+
+    # You will be able to load one or the other configurations in the following list with
+    # data = datasets.load_dataset('my_dataset', 'first_domain')
+    # data = datasets.load_dataset('my_dataset', 'second_domain')
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(
+            name="multilabel",
+            version=VERSION,
+            description="""
+            Full multilabel dataset with target values ranging
+            from 0 to 3 representing the votes from each annotator.
+            """,
+        ),
+        datasets.BuilderConfig(
+            name="binary",
+            version=VERSION,
+            description="""
+            Binary classification dataset version separated in train, dev and test test.
+            A text is considered toxic if at least one of the multilabel classes were labeled
+            by at least one annotator.
+            """,
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = "binary"
+
+    def _info(self):
+        if self.config.name == "binary":
+            features = datasets.Features(
+                {
+                    "text": datasets.Value("string"),
+                    "toxic": datasets.Value("uint32"),
+                }
+            )
+        else:
+            features = datasets.Features(
+                {
+                    "text": datasets.Value("string"),
+                    "homophobia": datasets.Value("uint32"),
+                    "obscene": datasets.Value("uint32"),
+                    "insult": datasets.Value("uint32"),
+                    "racism": datasets.Value("uint32"),
+                    "misogyny": datasets.Value("uint32"),
+                    "xenophobia": datasets.Value("uint32"),
```

Comment:
Same here, you can use `datasets.ClassLabel(names=["false", "true"])` for example

Reply:
Hey @lhoestq,

I tried

```
features = datasets.Features(
                {
                    "text": datasets.Value("string"),
                    "homophobia": datasets.ClassLabel(names=["not-homophobia", "homophobia"]),
                    "obscene": datasets.ClassLabel(names=["not-obscene", "obscene"]),
                    "insult": datasets.ClassLabel(names=["not-insult", "insult"]),
                    "racism": datasets.ClassLabel(names=["not-racism", "racism"]),
                    "misogyny": datasets.ClassLabel(names=["not-misogyny", "misogyny"]),
                    "xenophobia": datasets.ClassLabel(names=["not-xenophobia", "xenophobia"]),
                }
```

But when testing, I get this error message: ValueError: Class label 2 greater than configured num_classes 2

Reply:
I think I got it. That's because these aren't binary (true or false), but rather the amount of votes from the three annotators to that class. I coded it this way:

```
features = datasets.Features(
                {
                    "text": datasets.Value("string"),
                    "homophobia": datasets.ClassLabel(names=["zero_votes", "one_vote", "two_votes", "three_votes"]),
                    "obscene": datasets.ClassLabel(names=["zero_votes", "one_vote", "two_votes", "three_votes"]),
                    "insult": datasets.ClassLabel(names=["zero_votes", "one_vote", "two_votes", "three_votes"]),
                    "racism": datasets.ClassLabel(names=["zero_votes", "one_vote", "two_votes", "three_votes"]),
                    "misogyny": datasets.ClassLabel(names=["zero_votes", "one_vote", "two_votes", "three_votes"]),
                    "xenophobia": datasets.ClassLabel(names=["zero_votes", "one_vote", "two_votes", "three_votes"]),
                }
            )
```

Reply:
Oh yes indeed ! My bad ^^

That's good this way, thanks